src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -917,4 +917,64 @@
       it { expect(package.attribs.where(attrib_type: attribute_type)).to be_empty }
     end
   end
+
+  describe 'POST #new_release_request' do
+    let(:project) { create(:maintenance_incident_project) }
+
+    before do
+      login user
+    end
+
+    context 'with skiprequest param' do
+      before do
+        post :new_release_request, params: { project: project, skiprequest: true }
+      end
+
+      it { expect(response).to redirect_to( project_show_path(project)) }
+    end
+
+    context 'when raises an APIException' do
+      before do
+        post :new_release_request, params: { project: project }
+      end
+
+      it { expect(flash[:error]).to eq 'Internal problem while release request creation' }
+      it { expect(response).to redirect_to( project_show_path(project)) }
+    end
+
+    shared_examples 'a non APIException' do |exception_class|
+      before do
+        allow_any_instance_of(BsRequest).to receive(:save!).and_raise(exception_class, "boom #{exception_class}")
+        post :new_release_request, params: { project: project }
+      end
+
+      it { expect(flash[:error]).to eq "boom #{exception_class}" }
+      it { expect(response).to redirect_to( project_show_path(project)) }
+    end
+
+    context 'when raises a non APIException' do
+      [Patchinfo::IncompletePatchinfo,
+       BsRequestAction::UnknownProject,
+       BsRequestAction::BuildNotFinished,
+       BsRequestActionMaintenanceRelease::RepositoryWithoutReleaseTarget,
+       BsRequestActionMaintenanceRelease::RepositoryWithoutArchitecture,
+       BsRequestActionMaintenanceRelease::ArchitectureOrderMissmatch,
+       BsRequestAction::VersionReleaseDiffers,
+       BsRequestAction::UnknownTargetProject,
+       BsRequestAction::UnknownTargetPackage].each do |exception_class|
+        it_behaves_like 'a non APIException', exception_class
+      end
+    end
+
+    context 'when is successful' do
+      before do
+        allow_any_instance_of(BsRequest).to receive(:save!).and_return(true)
+        allow_any_instance_of(BsRequest).to receive(:number).and_return(1)
+        post :new_release_request, params: { project: project }
```

Comment:
I agree with you that the lines of creating an instance of a `BsRequest` class with an `BsRequestActionMaintenanceRelease` action must be tested. Also there are no more places in the the application where you can see a creation of a `BsRequest` with a `BsRequestActionMaintenanceRelease`.

On the other hand, all the lines of the `new_release_request` acion are covered, and in a strict way, the test of this action of the `project` controller is finished.

Can we agree on closing this pull request, and create other pull request for testing `BsRequest` and `BsRequestActionMaintenanceRelease` models? There are no model tests for them now.